src/core/components/files-regular/add-from-url.js

```diff
@@ -1,41 +1,22 @@
 'use strict'
 
 const { URL } = require('iso-url')
-const fetch = require('../../runtime/fetch-nodejs')
-
-module.exports = (self) => {
-  return async (url, options, callback) => {
-    if (typeof options === 'function') {
-      callback = options
-      options = {}
-    }
-
-    let files
-
-    try {
-      const parsedUrl = new URL(url)
-      const res = await fetch(url)
-
-      if (!res.ok) {
-        throw new Error('unexpected status code: ' + res.status)
-      }
-
-      // TODO: use res.body when supported
-      const content = Buffer.from(await res.arrayBuffer())
-      const path = decodeURIComponent(parsedUrl.pathname.split('/').pop())
-
-      files = await self.add({ content, path }, options)
-    } catch (err) {
-      if (callback) {
-        return callback(err)
-      }
-      throw err
-    }
+const nodeify = require('promise-nodeify')
+const { default: ky } = require('ky-universal')
+
+module.exports = (ipfs) => {
+  const addFromURL = async (url, opts = {}) => {
+    const res = await ky.get(url)
+    const path = decodeURIComponent(new URL(res.url).pathname.split('/').pop())
+    const content = Buffer.from(await res.arrayBuffer())
```

Comment:
One day we will use something like https://github.com/ipfs/js-ipfs-http-client/blob/master/src/lib/stream-to-iterable.js to stream the response.

Reply:
That day is today.  Well, yesterday.  Since we merged #2379 you should be able to pass iterables to `ipfs.add`.

Reply:
Seems that `ky` does not expose ReadableStream yet:
https://github.com/sindresorhus/ky/issues/3#issuecomment-502414532

To avoid blocking this PR, I'd keep this as-is and add streaming in separate PR, after support in `ky` lands.